tributors/utils/prompt.py

```diff
@@ -27,3 +27,32 @@ def request_input():
             lines.append(line)
 
     return "\n".join(lines)
+
+
+def choice_prompt(prompt, choices, choice_prefix=None, multiple=False):
+    """Ask the user for a prompt, and only return when one of the requested
+       options is provided.
+       Parameters
+       ==========
+       prompt: the prompt to ask the user
+       choices: a list of choices that are valid.   
+       multiple: allow multiple responses (separated by spaces)
+    """
+    choice = None
+    print(prompt)
+    get_input = getattr(__builtins__, "raw_input", input)
+
+    if not choice_prefix:
+        choice_prefix = "/".join(choices)
+    message = "[%s] : " % (choice_prefix)
+
+    while choice not in choices:
+        choice = get_input(message).strip()
+
+        # If multiple allowed, add selection to choices if includes all vaid
+        if multiple is True:
+            contenders = choice.strip().split(" ")
+            if all([x in choices for x in contenders]):
+                choices.append(choice)
+        message = "Please enter a valid option in [%s]" % (choice_prefix)
```

Comment:
nit picking: `(choice_prefix)` isn't a tuple, so if `choice_prefix` ends up being a `dict` -- it would blow.  You could either explicitly `str` it or ensure a tuple via `(choice_prefix,)`

Reply:
Why not just do `% choice_prefix` ?

Reply:
should be as good here if we are certain it is a `str`.  I am sometimes just over-defensive I guess ;)

Reply:
That's not so bad! I fixed it to not be tupley :)